translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java

```diff
@@ -0,0 +1,363 @@
+        Date date = new Date();
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
+        stepData.put("KuraDataMessage", kuraDataMessage);
+    }
+
+    @And("^I try to translate kura data message to mqtt message$")
+    public void iTryToTranslateKuraDataMessageToMqttMessage() throws Exception {
+        KuraDataMessage kuraDataMessage = (KuraDataMessage) stepData.get("KuraDataMessage");
+        MqttMessage mqttMessage = translatorDataKuraMqtt.translate(kuraDataMessage);
+        stepData.put("MqttMessage", mqttMessage);
+    }
+
+    @Then("^I got mqtt message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and (?:empty body|non empty body)$")
+    public void iGotMqttMessageWithChannelWithClientIdScopeAndWithoutMetricsAndBodyInPayload(String scope, String clientId) {
+        MqttMessage mqttMessage = (MqttMessage) stepData.get("MqttMessage");
+
+        String requestTopic = scope.concat("/" + clientId);
+        assertEquals(requestTopic, mqttMessage.getRequestTopic().getTopic());
+
+        if (mqttMessage.getPayload().getBody().length == 0) {
+            assertTrue(mqttMessage.getPayload().getBody().length == 0);
+        } else {
+            assertTrue(mqttMessage.getPayload().getBody().length != 0);
+        }
+    }
+
+    @And("^I got kura response message channel with \"(.+)\", \"(.+)\", \"(.+)\", \"(.+)\", \"(.+)\" and \"(.+)\" data$")
+    public void iGotKuraResponseMessageChannelWithAndData(String replyPart, String requestId, String appId, String messageClassification, String scope, String clientId) {
+        KuraResponseMessage kuraResponseMessage = (KuraResponseMessage) stepData.get("KuraResponseMessage");
+
+        assertTrue(kuraResponseMessage.getChannel().getReplyPart().equals(replyPart));
+        assertTrue(kuraResponseMessage.getChannel().getRequestId().equals(requestId));
+        assertTrue(kuraResponseMessage.getChannel().getAppId().equals(appId));
+        assertTrue(kuraResponseMessage.getChannel().getMessageClassification().equals(messageClassification));
+        assertTrue(kuraResponseMessage.getChannel().getScope().equals(scope));
+        assertTrue(kuraResponseMessage.getChannel().getClientId().equals(clientId));
+    }
+
+    @Given("^I create kura data message with channel with scope \"([^\"]*)\", client id \"([^\"]*)\" and payload without body and with metrics$")
+    public void iCreateKuraDataMessageWithChannelWithScopeClientIdAndPayloadWithoutBodyAndWithMetrics(String scope, String clientId) {
+        Date date = new Date();
+        KuraDataChannel kuraDataChannel = new KuraDataChannel(scope, clientId);
+        KuraDataPayload kuraDataPayload = new KuraDataPayload();
+        kuraDataPayload.getMetrics().put("response.code", 200);
+        KuraDataMessage kuraDataMessage = new KuraDataMessage(kuraDataChannel, date, kuraDataPayload);
```

Comment:
insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.